components/BottomNavigation/src/MDCBottomNavigationBarController.m

```diff
@@ -335,9 +347,56 @@ - (void)addContentViewController:(UIViewController *)viewController {
   if (viewController && doesNotContainViewController) {
     [self addChildViewController:viewController];
     [self.content addSubview:viewController.view];
-    [self addConstraintsForContentView:viewController.view];
+    [self addConstraintsForChildViewControllerView:viewController.view];
     [viewController didMoveToParentViewController:self];
   }
+  [self updateNavigationBarInsets];
+}
+
+- (void)viewDidLayoutSubviews {
+  [super viewDidLayoutSubviews];
+
+  [self updateNavigationBarInsets];
+}
+
+- (void)updateNavigationBarInsets {
+  UIEdgeInsets currentSafeAreaInsets = UIEdgeInsetsZero;
+  CGFloat navigationBarHeight = CGRectGetHeight(self.navigationBar.frame);
+  if (@available(iOS 11.0, *)) {
+    currentSafeAreaInsets = self.view.safeAreaInsets;
+  }
+  UIEdgeInsets additionalSafeAreaInsets =
+      UIEdgeInsetsMake(0, 0, navigationBarHeight - currentSafeAreaInsets.bottom, 0);
+  if (@available(iOS 11.0, *)) {
+    self.selectedViewController.additionalSafeAreaInsets = additionalSafeAreaInsets;
+  } else {
+    self.content.layoutMargins = additionalSafeAreaInsets;
+    // Based on an article by Ryan Fox, attempt to adjust the contentInset of either the view
+    // or its first subview.
+    // https://medium.com/@wailord/the-automaticallyadjustsscrollviewinsets-rabbit-hole-b9153a769ce9
+    if (self.selectedViewController.automaticallyAdjustsScrollViewInsets) {
+      UIScrollView *scrollView;
+      if ([self.selectedViewController.view isKindOfClass:[UIScrollView class]]) {
+        scrollView = (UIScrollView *)self.selectedViewController.view;
+      } else if ([self.selectedViewController.view.subviews.firstObject
+                     isKindOfClass:[UIScrollView class]]) {
+        scrollView = (UIScrollView *)self.selectedViewController.view.subviews.firstObject;
+      }
+      // Ideally, we would probably set an associated object that tracks the insets applied
```

Comment:
should this be explained in a bug instead and linked here? it seems like you are not fully satisfied with the solution and would want a follow up.

Reply:
I don't think it necessitates follow-up work (if I did I would have simply done the work).  I would expect that a BottomNavigationController be a root view controller, and so another container view controller won't be manipulating the `contentInset` property. However, if we find a situation where that assumption is unreasonable, then adding the complexity/cost of associated objects to track the added insets would be warranted.  I expect in most cases where that complexity exists, a custom UIViewController that uses the Bottom Navigation Bar (view) would be a better solution anyway.